src/Type/ObjectType.php

```diff
@@ -248,16 +248,22 @@ public function accepts(Type $type, bool $strictTypes): TrinaryLogic
 			return TrinaryLogic::createMaybe();
 		}
 
-		if (!$type instanceof TypeWithClassName) {
+		$thatClassNames = $type->getObjectClassNames();
+		if (count($thatClassNames) > 1) {
+			throw new ShouldNotHappenException();
```

Comment:
Why it couldn't be more than one? In the world where the only CompoundType is NeverType, this should be supported and implemented. By calling `checkSubclassAcceptability` for each class name and combining the results correctly.

Reply:
I even had something like that already but removed it again 😅 the problem for me was that I couldn't come up with a test (because as you hinted already, CompoundType is already handled before) and I was afraid that this "dead" code might break at some point in the future in weird ways. That's why I thought it might make sense to just not support it for now and break in more expected ways

Reply:
That's fine - write something that works, try it with a unit test and maybe temporarily make UnionType not implement CompoundType to see if your logic works...

When I make most types no longer implement CompoundType, I don't want to hunt for ShouldNotHappenExceptions 😊

Reply:
this is a good example, I tried to
> By calling checkSubclassAcceptability for each class name and combining the results correctly. 

added test cases and commented out the CompoundType handling. I either had
```php
return TrinaryLogic::createNo()->lazyOr(
	$thatClassNames,
	fn (string $thatClassName) => $this->checkSubclassAcceptability($thatClassName),
);
```

or 

```php
return TrinaryLogic::createYes()->lazyAnd(
	$thatClassNames,
	fn (string $thatClassName) => $this->checkSubclassAcceptability($thatClassName),
);
```

but both are wrong. I don't know anymore if it was a union or an intersection, so I have no clue how to deal with them :/

my added test cases for `accepts()` were
```php
			[
				new ObjectType(DateTimeInterface::class),
				new UnionType([
					new ObjectType(stdClass::class),
					new ObjectType(stdClass::class),
				]),
				TrinaryLogic::createNo(),
			],
			[
				new ObjectType(DateTimeInterface::class),
				new UnionType([
					new ObjectType(stdClass::class),
					new ObjectType(DateTime::class),
				]),
				TrinaryLogic::createMaybe(), // was no with lazyAnd(), was yes with lazyOr()
			],
			[
				new ObjectType(DateTimeInterface::class),
				new UnionType([
					new ObjectType(DateTime::class),
					new ObjectType(DateTime::class),
				]),
				TrinaryLogic::createYes(),
			],
			[
				new ObjectType(DateTimeInterface::class),
				new IntersectionType([
					new ObjectType(stdClass::class),
					new ObjectType(stdClass::class),
				]),
				TrinaryLogic::createNo(),
			],
			[
				new ObjectType(DateTimeInterface::class),
				new IntersectionType([
					new ObjectType(stdClass::class),
					new ObjectType(DateTime::class),
				]),
				TrinaryLogic::createYes(), // was no with lazyAnd()
			],
			[
				new ObjectType(DateTimeInterface::class),
				new IntersectionType([
					new ObjectType(DateTime::class),
					new ObjectType(DateTime::class),
				]),
				TrinaryLogic::createYes(),
			],
```

these are my main problems here.. any ideas?